packages/lib/server/getLuckyUser.ts

```diff
@@ -455,6 +526,34 @@ export function getLuckyUser_requiresDataToBePreFetched<
   };
 }
 
+function mergeOverlappingRanges(ranges: { start: Date; end: Date }[]): { start: Date; end: Date }[] {
+  if (ranges.length === 0) return [];
+
+  const sortedRanges = ranges.sort((a, b) => a.start.valueOf() - b.start.valueOf());
+
+  const mergedRanges: { start: Date; end: Date }[] = [sortedRanges[0]];
+
+  for (let i = 1; i < sortedRanges.length; i++) {
+    const lastMergedRange = mergedRanges[mergedRanges.length - 1];
+    const currentRange = sortedRanges[i];
+
+    if (currentRange.start.getTime() <= lastMergedRange.end.getTime()) {
+      lastMergedRange.end = new Date(Math.max(lastMergedRange.end.getTime(), currentRange.end.getTime()));
+    } else {
+      mergedRanges.push(currentRange);
+    }
+  }
+  return mergedRanges;
+}
+
+function isFullDayEvent(date1: Date, date2: Date) {
```

Comment:
```suggestion
function isFullDayEvent(date1: Date, date2: Date) {
  const MILLISECONDS_IN_A_DAY = 24 * 60 * 60 * 1000;
  const difference = Math.abs(date1.getTime() - date2.getTime());

  // Check if the time difference is approximately one full day
  return Math.abs(difference - MILLISECONDS_IN_A_DAY) < 1000;
```

Just to prevent a case where we're like 365ms off and it doesn't see it as a full day - decrease precision.